src/rgw/rgw_reshard.cc

```diff
@@ -664,16 +636,13 @@ int RGWBucketReshard::do_reshard(int num_shards,
     return -EIO;
   }
 
-  ret = store->ctl()->bucket->link_bucket(new_bucket_info.owner, new_bucket_info.bucket, bucket_info.creation_time, null_yield);
+  //overwrite current_index for the next reshard process
+  bucket_info.layout.current_index = *bucket_info.layout.target_index;
+  bucket_info.layout.resharding = rgw::BucketReshardState::NONE;
```

Comment:
should also clear the target index here with `bucket_info.layout.target_index.reset()` or `target_index = std::nullopt;`

Reply:
done